### PR TITLE
FOGL-858 (Part-A)

### PR DESCRIPTION
**Pull request by ashish-jabble**
- [x] audit.py
- [x] backup_restore.py
- [x] browser.py
- [ ] common.py --- **_No ACTION required_**
- [x] configuration.py
- [ ] scheduler.py -- **To avoid conflicts, will take up later once https://github.com/foglamp/FogLAMP/pull/411 merged**
- [x] statistics.py

**TESTS O/P**
- [x] test_audit.py
```
collected 22 items 

test_audit.py::TestAudit::test_get_severity PASSED
test_audit.py::TestAudit::test_get_log_codes PASSED
test_audit.py::TestAudit::test_get_audit_with_params[-6-6] PASSED
test_audit.py::TestAudit::test_get_audit_with_params[?skip=1-6-5] PASSED
test_audit.py::TestAudit::test_get_audit_with_params[?source=PURGE-4-4] PASSED
test_audit.py::TestAudit::test_get_audit_with_params[?source=PURGE&severity=error-3-3] PASSED
test_audit.py::TestAudit::test_get_audit_with_params[?source=PURGE&severity=ERROR&limit=1-3-1] PASSED
test_audit.py::TestAudit::test_get_audit_with_params[?source=PURGE&severity=INFORMATION&limit=1&skip=1-1-0] PASSED
test_audit.py::TestAudit::test_get_audit_with_params[?source=LOGGN&severity=FATAL-0-0] PASSED
test_audit.py::TestAudit::test_get_audit_with_params[?source=&severity=&limit=&skip=-6-6] PASSED
test_audit.py::TestAudit::test_params_with_bad_data[?limit=invalid-400-Limit must be a positive integer] PASSED
test_audit.py::TestAudit::test_params_with_bad_data[?limit=-1-400-Limit must be a positive integer] PASSED
test_audit.py::TestAudit::test_params_with_bad_data[?skip=invalid-400-Skip/Offset must be a positive integer] PASSED
test_audit.py::TestAudit::test_params_with_bad_data[?skip=-1-400-Skip/Offset must be a positive integer] PASSED
test_audit.py::TestAudit::test_params_with_bad_data[?severity=BLA-400-'BLA' is not a valid severity] PASSED
test_audit.py::TestAudit::test_params_with_bad_data[?source=blah-400-blah is not a valid source] PASSED
test_audit.py::TestAudit::test_post_audit SKIPPED
test_audit.py::TestAudit::test_get_all_notifications SKIPPED
test_audit.py::TestAudit::test_get_notification SKIPPED
test_audit.py::TestAudit::test_post_notification SKIPPED
test_audit.py::TestAudit::test_update_notification SKIPPED
test_audit.py::TestAudit::test_delete_notification SKIPPED

===== 16 passed, 6 skipped in 0.67 seconds ====
```
- [x] test_backup_restore.py
```
collected 19 items 

test_backup_restore.py::TestBackup::test_get_backups[-4-exp_output0] PASSED
test_backup_restore.py::TestBackup::test_get_backups[?limit=1-1-exp_output1] PASSED
test_backup_restore.py::TestBackup::test_get_backups[?skip=3-1-exp_output2] PASSED
test_backup_restore.py::TestBackup::test_get_backups[?limit=2&skip=1-2-exp_output3] PASSED
test_backup_restore.py::TestBackup::test_get_backups[?status=failed-1-exp_output4] PASSED
test_backup_restore.py::TestBackup::test_get_backups[?limit=2&skip=1&status=completed-1-exp_output5] PASSED
test_backup_restore.py::TestBackup::test_get_backups[?limit=&skip=&status=-4-exp_output6] PASSED
test_backup_restore.py::TestBackup::test_get_backups_invalid[?limit=invalid-400-Limit must be a positive integer] PASSED
test_backup_restore.py::TestBackup::test_get_backups_invalid[?limit=-10-400-Limit must be a positive integer] PASSED
test_backup_restore.py::TestBackup::test_get_backups_invalid[?skip=invalid-400-Skip/Offset must be a positive integer] PASSED
test_backup_restore.py::TestBackup::test_get_backups_invalid[?skip=-1-400-Skip/Offset must be a positive integer] PASSED
test_backup_restore.py::TestBackup::test_get_backups_invalid[?status=invalid-400-'INVALID' is not a valid status] PASSED
test_backup_restore.py::TestBackup::test_get_backup_details[request_params0-output0] PASSED
test_backup_restore.py::TestBackup::test_get_backup_details_invalid[invalid-400-Invalid backup id] PASSED
test_backup_restore.py::TestBackup::test_get_backup_details_invalid[-1-404-Backup with -1 does not exist] PASSED
test_backup_restore.py::TestBackup::test_create_backup SKIPPED
test_backup_restore.py::TestBackup::test_delete_backup SKIPPED
test_backup_restore.py::TestBackup::test_get_backup_status PASSED
test_backup_restore.py::TestRestore::test_restore SKIPPED

=== 16 passed, 3 skipped in 0.41 seconds ===
```
- [x] test_browser_assets.py
```
collected 38 items 

test_browser_assets.py::TestBrowseAssets::test_get_all_assets PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_readings PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_readings_q_limit PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_readings_q_sec PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_readings_q_min PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_readings_q_hrs PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_readings_q_time_complex PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_with_empty_params PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_q_limit PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_q_sec PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_q_min PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_q_hrs PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_q_time_complex PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_stats xfail
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_stats_q_sec xfail
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_stats_q_min xfail
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_stats_q_hrs xfail
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_stats_q_time_complex xfail
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_time_avg xfail
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_invalid_group PASSED
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_time_avg_q_group_sec xfail
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_time_avg_q_group_min xfail
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_time_avg_q_group_hrs xfail
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_time_avg_q_limit_group_hrs xfail
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_time_avg_q_time xfail
test_browser_assets.py::TestBrowseAssets::test_get_asset_sensor_readings_time_avg_q_group_time_limit xfail
test_browser_assets.py::TestBrowseAssets::test_params_with_bad_data[?limit=invalid-400-Limit must be a positive integer] PASSED
test_browser_assets.py::TestBrowseAssets::test_params_with_bad_data[?limit=-1-400-Limit must be a positive integer] PASSED
test_browser_assets.py::TestBrowseAssets::test_params_with_bad_data[?skip=invalid-400-Skip/Offset must be a positive integer] PASSED
test_browser_assets.py::TestBrowseAssets::test_params_with_bad_data[?skip=-1-400-Skip/Offset must be a positive integer] PASSED
test_browser_assets.py::TestBrowseAssets::test_params_with_bad_data[?minutes=-1-400-Time must be a positive integer] PASSED
test_browser_assets.py::TestBrowseAssets::test_params_with_bad_data[?minutes=blah-400-Time must be a positive integer] PASSED
test_browser_assets.py::TestBrowseAssets::test_params_with_bad_data[?seconds=-1-400-Time must be a positive integer] PASSED
test_browser_assets.py::TestBrowseAssets::test_params_with_bad_data[?seconds=blah-400-Time must be a positive integer] PASSED
test_browser_assets.py::TestBrowseAssets::test_params_with_bad_data[?hours=-1-400-Time must be a positive integer] PASSED
test_browser_assets.py::TestBrowseAssets::test_params_with_bad_data[?hours=blah-400-Time must be a positive integer] PASSED
test_browser_assets.py::TestBrowseAssets::test_error_when_no_rows_key_available PASSED

==== 26 passed, 12 xfailed in 0.96 seconds ====
```
- [x] test_configuration.py
```
collected 10 items 

test_configuration.py::TestConfigMgr::test_get_categories PASSED
test_configuration.py::TestConfigMgr::test_get_category PASSED
test_configuration.py::TestConfigMgr::test_get_invalid_category PASSED
test_configuration.py::TestConfigMgr::test_get_invalid_category_item SKIPPED
test_configuration.py::TestConfigMgr::test_get_category_item PASSED
test_configuration.py::TestConfigMgr::test_set_category_item_value PASSED
test_configuration.py::TestConfigMgr::test_edit_category_item_value PASSED
test_configuration.py::TestConfigMgr::test_unset_config_item PASSED
test_configuration.py::TestConfigMgr::test_merge_category SKIPPED
test_configuration.py::TestConfigMgr::test_check_error_code_message PASSED

===== 8 passed, 2 skipped in 0.19 seconds ====
```
- [x] test_statistics_history.py
```
collected 4 items 

test_statistics_history.py::TestStatisticsHistory::test_get_statistics_history PASSED
test_statistics_history.py::TestStatisticsHistory::test_get_statistics_history_with_limit PASSED
test_statistics_history.py::TestStatisticsHistory::test_get_statistics_history_limit_with_bad_data[?limit=invalid-400-Limit must be a positive integer] PASSED
test_statistics_history.py::TestStatisticsHistory::test_get_statistics_history_limit_with_bad_data[?limit=-1-400-Limit must be a positive integer] PASSED

=== 4 passed in 45.30 seconds ===
```
- [x] test_statistics.py
```
collected 3 items 

test_statistics.py::TestStatistics::test_get_statistics PASSED
test_statistics.py::TestStatistics::test_get_updated_statistics PASSED
test_statistics.py::TestStatistics::test_get_statistics_with_new_key_entry PASSED

==== 3 passed in 0.20 seconds ====
```

### CODE REVIEW

**Comment by ashish-jabble** [comment]
@praveen-garg @amarendra-dianomic Ready for review.

File: tests/unit-tests/python/foglamp_test/services/core/api/test_statistics.py [hunk 1]
```diff
@@ -56,8 +56,8 @@ def teardown_class(cls):
         # curl -X GET http://localhost:8081/foglamp/statistics
         r = requests.get(BASE_URL+'/statistics')
         res = r.json()
-        assert 200 == r.status_code
-        assert 9 == len(res)
+        assert r.status_code == 200
```
Comment (amarendra-dianomic) [inline tests/unit-tests/python/foglamp_test/services/core/api/test_statistics.py]: should it not be expected == actual?

File: tests/unit-tests/python/foglamp_test/services/core/api/test_statistics.py [hunk 2]
```diff
@@ -102,8 +102,8 @@ def teardown_class(cls):
         r = requests.get(BASE_URL + '/statistics')
         res = r.json()
 
-        assert 200 == r.status_code
-        assert 9 == len(res)
+        assert r.status_code == 200
```
Comment (amarendra-dianomic) [inline tests/unit-tests/python/foglamp_test/services/core/api/test_statistics.py]: should it not be expected == actual?

File: tests/unit-tests/python/foglamp_test/services/core/api/test_statistics.py [hunk 3]
```diff
@@ -116,8 +116,8 @@ def teardown_class(cls):
         r = requests.get(BASE_URL + '/statistics')
         res = r.json()
 
-        assert 200 == r.status_code
-        assert 10 == len(res)
+        assert r.status_code == 200
```
Comment (amarendra-dianomic) [inline tests/unit-tests/python/foglamp_test/services/core/api/test_statistics.py]: should it not be expected == actual?

File: tests/unit-tests/python/foglamp_test/services/core/api/test_browser_assets.py [hunk 4]
```diff
@@ -129,45 +135,45 @@ def group_date_time(self, unit=None):
         conn = http.client.HTTPConnection(BASE_URL)
         conn.request("GET", '/foglamp/asset')
         r = conn.getresponse()
-        assert 200 == r.status
+        assert r.status == 200
```
Comment (amarendra-dianomic) [inline tests/unit-tests/python/foglamp_test/services/core/api/test_browser_assets.py]: should it not be expected == actual?

File: tests/unit-tests/python/foglamp_test/services/core/api/test_browser_assets.py [hunk 5]
```diff
@@ -129,45 +135,45 @@ def group_date_time(self, unit=None):
...
         """
         Verify that if more than 20 readings, only 20 are returned as the default limit for asset_code
-        http://localhost:8082/foglamp/asset/TESTAPI
+        http://localhost:8081/foglamp/asset/TESTAPI
         """
         conn = http.client.HTTPConnection(BASE_URL)
         conn.request("GET", '/foglamp/asset/{}'.format(test_data_asset_code))
         r = conn.getresponse()
-        assert 200 == r.status
+        assert r.status == 200
```
Comment (amarendra-dianomic) [inline tests/unit-tests/python/foglamp_test/services/core/api/test_browser_assets.py]: should it not be expected == actual?

File: tests/unit-tests/python/foglamp_test/services/core/api/test_browser_assets.py [hunk 6]
```diff
@@ -129,45 +135,45 @@ def group_date_time(self, unit=None):
...
         r = r.read().decode()
         conn.close()
         retval = json.loads(r)
-        assert 20 == len(retval)
+        assert len(retval) == 20
```
Comment (amarendra-dianomic) [inline tests/unit-tests/python/foglamp_test/services/core/api/test_browser_assets.py]: should it not be expected == actual?

File: tests/unit-tests/python/foglamp_test/services/core/api/test_browser_assets.py (hunk 5 above)
Comment (amarendra-dianomic) [inline tests/unit-tests/python/foglamp_test/services/core/api/test_browser_assets.py]: should it not be expected == actual?

**Comment by ashish-jabble** [comment]
@amarendra-dianomic As we agreed earlier use any style, a file can have any of this style but must be consistent.

**Pull request merged**